### PR TITLE
add and use a helper for opening modals

### PR DESCRIPTION
**Pull request by sandlerr**
### Screenshots

screen shot 2016-10-10 at 3 21 47 pm
screen shot 2016-10-10 at 3 22 08 pm

@zendesk/vegemite

### CODE REVIEW

**Comment by adammw** [comment]
screenshot? otherwise the code looks legit...

**Comment by sandlerr** [comment]
added

File: lib/javascripts/base_app.js [hunk 1]
```diff
@@ -120,7 +120,7 @@ BaseApp.prototype = {
   promise: function(fn) {
     if (!_.isFunction(fn)) { throw new Error('`promise` needs to be passed a Function'); }
     var dfd = $.Deferred();
-    setTimeout(fn.bind(this, dfd.resolve.bind(dfd), dfd.reject.bind(dfd)));
+    _.defer(fn.bind(this, dfd.resolve.bind(dfd), dfd.reject.bind(dfd)));
```
Comment (svizzari) [inline lib/javascripts/base_app.js]: ❓

Comment (sandlerr) [inline lib/javascripts/base_app.js]: just convention

File: lib/javascripts/base_app.js [hunk 2]
```diff
@@ -144,6 +144,10 @@ BaseApp.prototype = {
     }
 
     this._storage.set(keyOrObject, value);
+  },
+
+  modal: function(url) {
```
Comment (danielbreves) [inline lib/javascripts/base_app.js]: This is the only method in this class that doesn't match framework v1 😕

Comment (sandlerr) [inline lib/javascripts/base_app.js]: is there a better place for it?

Comment (danielbreves) [inline lib/javascripts/base_app.js]: Maybe there's a way we can provide an actual shim, by extending jQuery and taking a template name as an argument? Then the Base constructor could handle switching to the template if the location is modal? We could attempt to pass the data serialised as a url  parameter? This code is fine as an example implementation but I'm not sure it belongs here.

File: src/templates/modal.hdbs [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<p>Not feeling modal?</p>
```
Comment (adammw) [inline src/templates/modal.hdbs]: I don't get the joke...

Comment (sandlerr) [inline src/templates/modal.hdbs]: tough crowd

**Pull request merged**